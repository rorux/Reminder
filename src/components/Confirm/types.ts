import React from 'react';

export type TConfirmProps = {
  open: boolean;
  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
  handler: (event: React.MouseEvent<HTMLElement>, id: string) => void;
  id: string;
};
