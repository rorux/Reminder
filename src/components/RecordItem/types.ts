import React from 'react';
import { TRecordAccordion } from '@components/RecordsAccordion/types';

export type TRecordItemProps = {
  record: TRecordAccordion;
  handleChange: (panel: string) => (event: React.SyntheticEvent, isExpanded: boolean) => void;
  expanded: string | false;
  handleDelete: (event: React.MouseEvent<HTMLElement>, id: string) => void;
};
