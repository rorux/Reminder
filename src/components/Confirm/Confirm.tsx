import React, { useEffect } from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogTitle from '@mui/material/DialogTitle';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import { TConfirmProps } from './types';

const Confirm: React.FC<TConfirmProps> = ({ open, setOpen, handler, id }) => {
  const [openConfirm, setOpenConfirm] = React.useState(true);

  useEffect(() => {
    setOpenConfirm(open);
  }, [open]);

  const handleClose = () => {
    setOpen(false);
  };

  const handlerFunc = (e: React.MouseEvent<HTMLElement>, id: string) => {
    handler(e, id);
    setOpen(false);
  };

  return (
    <>
      <Dialog
        open={openConfirm}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogContent>
          <DialogContentText>Удалить напоминание?</DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button data-test="disagree" onClick={handleClose}>
            Отмена
          </Button>
          <Button data-test="agree" onClick={(e) => handlerFunc(e, id)} autoFocus>
            Удалить
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default Confirm;
