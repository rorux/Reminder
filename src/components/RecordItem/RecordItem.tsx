import React from 'react';
import { Redirect } from 'react-router-dom';
import Accordion from '@mui/material/Accordion';
import AccordionDetails from '@mui/material/AccordionDetails';
import AccordionSummary from '@mui/material/AccordionSummary';
import Typography from '@mui/material/Typography';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import Stack from '@mui/material/Stack';
import Button from '@mui/material/Button';
import Confirm from '@components/Confirm';
import { TRecordItemProps } from './types';

const RecordItem: React.FC<TRecordItemProps> = ({
  record,
  expanded,
  handleChange,
  handleDelete,
}) => {
  const [isOpenConfirm, setIsOpenConfirm] = React.useState(false);
  const [isEdit, setIsEdit] = React.useState(false);

  if (isEdit) return <Redirect to={`/edit/${record.id}`} />;

  return (
    <Accordion
      expanded={expanded === `panel${record.id}`}
      onChange={handleChange(`panel${record.id}`)}
      key={record.id}
    >
      <AccordionSummary
        expandIcon={<ExpandMoreIcon />}
        aria-controls={`panel${record.id}bh-content`}
        id={`panel${record.id}bh-header`}
      >
        <Typography sx={{ width: '33%', flexShrink: 0 }}>{record.period}</Typography>
        <Typography sx={{ color: 'text.secondary' }}>{record.title}</Typography>
      </AccordionSummary>
      <AccordionDetails>
        <Typography>{record.comment}</Typography>
        <Stack direction="row" spacing={2} sx={{ mt: 2 }}>
          <Button
            data-test="edit-button"
            variant="contained"
            color="warning"
            sx={{ boxShadow: 0 }}
            onClick={() => setIsEdit(true)}
          >
            Изменить
          </Button>
          <Button
            data-test="delete-button"
            variant="contained"
            color="error"
            sx={{ boxShadow: 0 }}
            onClick={() => setIsOpenConfirm(true)}
          >
            Удалить
          </Button>
          <Confirm
            open={isOpenConfirm}
            setOpen={setIsOpenConfirm}
            handler={handleDelete}
            id={record.id}
          />
        </Stack>
      </AccordionDetails>
    </Accordion>
  );
};

export default RecordItem;
