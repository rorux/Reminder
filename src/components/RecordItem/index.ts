import RecordItem from './RecordItem';

export default RecordItem;
