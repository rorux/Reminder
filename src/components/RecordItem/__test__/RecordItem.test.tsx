import React from 'react';
import { Redirect } from 'react-router-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme, { shallow } from 'enzyme';
import RecordItem from '../RecordItem';
import Confirm from '@components/Confirm';

Enzyme.configure({ adapter: new Adapter() });

const record = {
  period: 'period',
  title: 'title',
  id: 'id',
  comment: 'comment',
};

const setUp = () =>
  shallow(
    <RecordItem record={record} handleChange={jest.fn()} handleDelete={jest.fn()} expanded="123" />
  );
describe('RecordItem component', () => {
  let component: Enzyme.ShallowWrapper;

  beforeEach(() => {
    component = setUp();
  });

  it('Should render RecordItem component', () => {
    expect(component).toMatchSnapshot();
  });

  it('Should open Confirm when clicked delete-button', () => {
    expect(component.find(Confirm).prop('open')).toBeFalsy();
    // @ts-ignore
    component.find(`[data-test='delete-button']`).props().onClick();
    expect(component.find(Confirm).prop('open')).toBeTruthy();
  });

  it('Should redirect when clicked edit-button', () => {
    // @ts-ignore
    component.find(`[data-test='edit-button']`).props().onClick();
    expect(component.find(Redirect)).toHaveLength(1);
  });
});
