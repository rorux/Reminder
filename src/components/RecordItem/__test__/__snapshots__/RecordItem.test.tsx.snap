// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordItem component Should render RecordItem component 1`] = `
<ForwardRef(Accordion)
  expanded={false}
  key="id"
>
  <ForwardRef(AccordionSummary)
    aria-controls="panelidbh-content"
    expandIcon={<Memo />}
    id="panelidbh-header"
  >
    <ForwardRef(Typography)
      sx={
        Object {
          "flexShrink": 0,
          "width": "33%",
        }
      }
    >
      period
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      sx={
        Object {
          "color": "text.secondary",
        }
      }
    >
      title
    </ForwardRef(Typography)>
  </ForwardRef(AccordionSummary)>
  <ForwardRef(AccordionDetails)>
    <ForwardRef(Typography)>
      comment
    </ForwardRef(Typography)>
    <ForwardRef(Stack)
      direction="row"
      spacing={2}
      sx={
        Object {
          "mt": 2,
        }
      }
    >
      <ForwardRef(Button)
        color="warning"
        data-test="edit-button"
        onClick={[Function]}
        sx={
          Object {
            "boxShadow": 0,
          }
        }
        variant="contained"
      >
        Изменить
      </ForwardRef(Button)>
      <ForwardRef(Button)
        color="error"
        data-test="delete-button"
        onClick={[Function]}
        sx={
          Object {
            "boxShadow": 0,
          }
        }
        variant="contained"
      >
        Удалить
      </ForwardRef(Button)>
      <Confirm
        handler={[MockFunction]}
        id="id"
        open={false}
        setOpen={[Function]}
      />
    </ForwardRef(Stack)>
  </ForwardRef(AccordionDetails)>
</ForwardRef(Accordion)>
`;
